src/osd/osd_types.cc

```diff
@@ -4961,9 +5002,22 @@ void object_manifest_t::decode(bufferlist::iterator& bl)
 void object_manifest_t::dump(Formatter *f) const
 {
   f->dump_unsigned("type", type);
-  f->open_object_section("redirect_target");
-  redirect_target.dump(f);
-  f->close_section();
+  {
+    f->open_object_section("redirect_target");
+    redirect_target.dump(f);
+    f->close_section();
+  }
+  {
```

Comment:
let's do something a more strongly typed structure, like
<pre>
if (type == chunk) {
open_array_section("chunk_map");
for (auto& p : chunk_map) {
  open_object_section("chunk");
  dump_unsigned("offset", p.first);
  p.second.dump(f);
  close_section();
}
close_section();
</pre>